src/shared/regions/endpoints.ts

```diff
@@ -0,0 +1,113 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Parses the endpoints.json file into a usable structure
+
+export interface Endpoints {
+    partitions: Partition[]
+}
+
+export interface Partition {
+    id: string
+    name: string
+    regions: Region[]
+    services: Service[]
+}
+
+export interface Region {
+    id: string
+    description: string
+}
+
+export interface Service {
+    id: string
+    endpoints: ServiceEndpoints[]
+    isRegionalized?: boolean
+    partitionEndpoint?: string
+}
+
+export interface ServiceEndpoints {
+    regionId: string
+    data: any
+}
+
+// --- JSON Serialization Structures ---
+interface ManifestEndpoints {
+    partitions?: ManifestPartition[]
+}
+
+interface JsonStringMap<T> {
+    [key: string]: T
+}
+
+interface ManifestPartition {
+    partition: string
+    partitionName: string
+    regions?: JsonStringMap<ManifestRegion>
+    services?: JsonStringMap<ManifestService>
+}
+
+interface ManifestRegion {
+    description: string
+}
+
+interface ManifestService {
+    endpoints?: JsonStringMap<any>
+    isRegionalized?: boolean
+    partitionEndpoint?: string
+}
+
+// --- END JSON Serialization Structures ---
+
+export function loadEndpoints(json: string): Endpoints {
+    const manifestEndpoints = JSON.parse(json) as ManifestEndpoints
```

Comment:
try/catch the JSON parse

Reply:
Error handling is necessary for corrupt input. Right now I'm leaning towards logging and returning undefined if an error rises instead of relying on callers to do this. Will revise...